Test Nuitka support for package Six

Pihu1998 opened the pull request:
This is a test to check if PyPI package Six is supported by Nuitka

kayhayen commented on tests/standalone/six.py:
```diff
@@ -0,0 +1,30 @@
+
+MAXSIZE = sys.maxsize
+
+def test_dispatch_types(value):
```
The Nuitka test runner is executing these as programs, with CPython and Nuitka, and will check the outputs, and merely defining a function named `test_*` is not going to do anything to run it. You can put stuff in functions, but then you have to also call it. Most often for standalone tests, we do just put things on module level.

Pihu1998 commented:
@kayhayen  Please review the above changes and please inform if there is any room for further improvements.

kayhayen commented on tests/standalone/six.py:
```diff
@@ -0,0 +1,28 @@
+
+MAXSIZE = sys.maxsize
+
+def dispatch_types(value):
```
Well, not being named `test_*` still doesn't make the code being called, you realize that?

Pihu1998 commented:
@kayhayen Please review the above changes. I have included a main() function that calls all of the functions. Please inform if it can be improved further.

kayhayen commented on tests/standalone/six.py:
```diff
@@ -0,0 +1,58 @@
+
+def dispatch_types(value):
+   
+    if isinstance(value, six.integer_types):
```
This is not a big coverage of the six package, is it?

Pihu1998 replied:
Okay, I think I should expand the usage of Six specifically.

kayhayen commented on tests/standalone/six.py:
```diff
@@ -0,0 +1,58 @@
+from __future__ import absolute_import
```
You need that absolute_import only due to your strange naming of the test as "six", why not go with SixUsing.py to match what other tests have.

Pihu1998 replied:
Yeah, Sure!

kayhayen replied:
Won't you delete this file then, it seems you forgot about that?

kayhayen commented on tests/standalone/six.py:
```diff
@@ -0,0 +1,58 @@
+        print(value)
+        
+    elif isinstance(value, six.class_types):
+        value
```
That's not doing anything.

kayhayen commented on tests/standalone/six.py:
```diff
@@ -0,0 +1,58 @@
+        print(value)
+
+    elif isinstance(value, six.string_types):
+        print(value)
```
That's not distinguishable from above branch.

kayhayen commented on tests/standalone/six.py:
```diff
@@ -0,0 +1,58 @@
+        value
+        
+
+def import_module(name):
```
Why is this here?

Pihu1998 replied:
Oh sorry, my bad. I will remove it.

kayhayen commented on tests/standalone/six.py:
```diff
@@ -0,0 +1,58 @@
+    return sys.modules[name]
+    
+
+def add_doc(func, doc):
```
This is not using the doc parameter, what is it intended for?

kayhayen commented:
@Pihu1998 Also do not just push new code, instead also respond to review comments I make, I failed to notice this one had changed, but assumed you didn't work on it.

kayhayen commented:
Well, maybe I was not precise enough. Respond to Github review comments right after you addressed them. Otherwise respond to ask a question about the comment or an outright refutation of it. But do not acknowledge, promise to make changes, or apologize.

I would much rather prefer to see a comment mean there is new code, or information I sought, so I can resolve the issue. Now there will be comments with responses, and I am left guessing when it makes sense to look at the code. That was not the point of asking you to reply, just to have something there, but rather to learn about what you have addressed.

Pihu1998 commented:
Sorry for the late push of code. I was working on it. I have added some more usage of the package Six in the code and also changed the test file name to 'SixUsing.py'. Please review the above changes.

kayhayen commented on tests/standalone/SixUsing.py:
```diff
@@ -0,0 +1,41 @@
+    print(dispatch_types('hi'))
+    u='hi'
+    print(dispatch_types(u.encode('utf-8')))
+    Dict = {'Tim': 18,'Charlie':12,'Tiffany':22,'Robert':25} 
```
Just a minor quibble, don't name local variables this way, makes me think it's a class or when I see it used.

As for the prints, import the `print_function` from future, and add some identifying prefixes, e.g. `dispatch_types int:`, the so output is more human readable. It works the way you did it, but it's not nice.

kayhayen commented on tests/standalone/SixUsing.py:
```diff
@@ -0,0 +1,41 @@
+    return six.viewkeys(dictionary), six.viewvalues(dictionary), six.viewitems(dictionary)
+
+
+class TestAssertCountEqual(unittest.TestCase):
```
So, is this is a six extension for unit tests here?

kayhayen commented:
Unlikely to be finished.
